src/lib/services/project-service.ts

```diff
@@ -596,20 +617,61 @@ export default class ProjectService {
     async statusJob(): Promise<void> {
         const projects = await this.store.getAll();
 
+        const statusUpdates = await Promise.all(
+            projects.map((project) => this.getStatusUpdates(project.id)),
+        );
+
         await Promise.all(
-            projects.map((project) =>
-                this.calculateAverageTimeToProd(project.id),
-            ),
+            statusUpdates.map((statusUpdate) => {
+                return this.projectStatusStore.updateStatus(
+                    statusUpdate.projectId,
+                    statusUpdate.updates,
+                );
+            }),
         );
     }
 
-    async calculateAverageTimeToProd(projectId: string): Promise<number> {
+    async getStatusUpdates(projectId: string): Promise<ICalculateStatus> {
         // Get all features for project with type release
         const features = await this.featureToggleStore.getAll({
             type: 'release',
             project: projectId,
         });
 
+        const [createdCurrentWindow, createdPastWindow] = await Promise.all([
+            await this.featureToggleStore.getByDate({
+                project: projectId,
+                dateAccessor: 'created_at',
+                date: subDays(new Date(), 30).toISOString(),
```

Comment:
it's difficult to test it with the "new Date()" hardcoded. What about clock pattern where we inject time provider as a dependency? https://twitter.com/jarek000000/status/897470232159236096

Reply:
I would like to explore this option, but I don't think I can get it done before my vacation and I don't want it to block the feature.

Reply:
@kwasniew I've been using that approach for years, didn't even realise it had a name, I learned something, thank you!